app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -168,33 +180,132 @@ class PopupView : FrameLayout {
+                                popupFieldLabel.text = fieldLabelWithValidationError
+                                popupFieldLabel.setTextColor(Color.RED)
+                            } else {
+                                popupFieldLabel.text = popupField.label
+                                popupFieldLabel.setTextColor(oldColors)
+                            }
                         }
                     }
                 }
             } else {
                 popupFieldEditValue.visibility = View.GONE
+                popupFieldCvdValues.visibility = View.GONE
                 popupFieldValue.visibility = View.VISIBLE
             }
         }
+
+        /**
+         * Sets up spinner for PopupFields that have a CodedValueDomain.
+         */
+        private fun setUpSpinner(codedValueDomain: CodedValueDomain, popupField: PopupField) {
+            val codedValuesNames = mutableListOf<String>()
+            codedValueDomain.codedValues.forEach { codedValue -> codedValuesNames.add(codedValue.name) }
+            popupFieldCvdValues.adapter = ArrayAdapter(
+                binding.root.context,
+                android.R.layout.simple_spinner_dropdown_item,
+                codedValuesNames
+            )
+            val codedValuePosition = codedValueDomain.codedValues.indexOfFirst { codedValue ->
+                codedValue.code == popupManager.getFieldValue(popupField)
+            }
+            // set the PopupField value as selected in the spinner
+            popupFieldCvdValues.setSelection(codedValuePosition)
+            popupFieldCvdValues.onItemSelectedListener =
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onNothingSelected(parent: AdapterView<*>?) {
+                        TODO("Not yet implemented")
+                    }
+
+                    override fun onItemSelected(
+                        parent: AdapterView<*>?,
+                        view: View?,
+                        position: Int,
+                        id: Long
+                    ) {
+                        popupManager.updateValue(
+                            codedValueDomain.codedValues[position].code,
+                            popupField
+                        )
+                    }
+                }
+        }
+
+        /**
+         * Updates the value of the specified PopupField to the appropriately cast string value of
+         * the specified value
+         */
+        private fun updateValue(popupField: PopupField, newValue: String): ArcGISRuntimeException? {
+            var error: ArcGISRuntimeException? = null
+            when (popupManager.getFieldType(popupField)) {
+                Field.Type.SHORT -> error =
+                    if (newValue.toShortOrNull() != null) {
+                        popupManager.updateValue(newValue.toShort(), popupField)
+                    } else {
```

Comment:
Why the `else`? Shouldn't we just error if the value cannot be converted to short? Same for the other cases below.

Reply:
The `else` is to pass in the String as is to the `PopupManager` so we can get the error back to propagate to the user.